lib/promise.rb

```diff
@@ -25,11 +25,18 @@ def self.all(enumerable)
     Group.new(new, enumerable).promise
   end
 
-  def self.map_value(obj)
-    if obj.is_a?(Promise)
-      obj.then { |value| yield value }
-    else
-      yield obj
+  if Gem.ruby_version >= Gem::Version.new('2.6.0')
```

Comment:
Hmm... Sorry I'm not well around rubygems too 😓 , but … Depending to `Gem` in library code smells a bit dangerous.

Can't we use simpler way as below? 🤔 

```suggestion
  if Object.new.respond_to?(:then)
```

```suggestion
  if RUBY_VERSION > '2.6.0'
```

Reply:
String comparison for versions or unpadded numbers is a bad practice, since it will be wrong when comparing numbers with a different number of digits.  For instance, `'10.0.0' < '2.6.0'` would evaluate to `true`.

Rubygems has been builtin into ruby for long time, so there shouldn't be a danger with its availability.  `Gem::Version` is comparing versions semantically, so there shouldn't be a danger with correctness (in fact, it even gets something like `Gem::Version.new('2.6.0.alpha') < Gem::Version.new('2.6.0')` right by evaluating it to `true`).

Reply:
Sure! Thank you!